common/src/main/java/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -34,10 +35,14 @@ typealias PostReceiptDataSuccessCallback = (CustomerInfo, body: JSONObject) -> U
 typealias PostReceiptDataErrorCallback = (PurchasesError, shouldConsumePurchase: Boolean, body: JSONObject?) -> Unit
 /** @suppress */
 typealias IdentifyCallback = Pair<(CustomerInfo, Boolean) -> Unit, (PurchasesError) -> Unit>
+/** @suppress */
+typealias DiagnosticsCallback = Pair<(JSONObject) -> Unit, (PurchasesError, Boolean) -> Unit>
 
 class Backend(
     private val apiKey: String,
+    private val appConfig: AppConfig,
     private val dispatcher: Dispatcher,
+    private val diagnosticsDispatcher: Dispatcher,
```

Comment:
Note that both the backend request and the file operations are executed in the same dispatcher. I believe that's ok but lmk if you have any concerns.

Reply:
Yeah that makes sense to isolate all of diagnostics in the same dispatcher.